domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt

```diff
@@ -0,0 +1,305 @@
+import org.oppia.app.model.State
+import org.oppia.app.model.StringList
+import org.oppia.app.model.SubtitledHtml
+import org.oppia.util.data.DataProviders
+
+const val TEST_EXPLORATION_ID_0 = "test_exp_id_0"
+const val TEST_EXPLORATION_ID_1 = "test_exp_id_1"
+private const val EXPLORATION_DATA_PROVIDER_ID = "ExplorationDataProvider"
+
+/** Controller for retrieving an exploration. */
+@Singleton
+class ExplorationDataController @Inject constructor(
+  private val context: Context, private val dataProviders: DataProviders
+) {
+
+  /**
+   * Returns an  [Exploration] given an ID.
+   */
+  fun getExplorationById(id: String): LiveData<AsyncResult<Exploration>> {
+    if (id == TEST_EXPLORATION_ID_0) {
+      return dataProviders.convertToLiveData(
+        dataProviders.createInMemoryDataProviderAsync(
+          EXPLORATION_DATA_PROVIDER_ID, this::retrieveWelcomeExplorationAsync
+        )
+      )
+    }
+    if (id == TEST_EXPLORATION_ID_1) {
+      return dataProviders.convertToLiveData(
+        dataProviders.createInMemoryDataProviderAsync(
+          EXPLORATION_DATA_PROVIDER_ID, this::retrieveAbboutOppiaExplorationAsync
+        )
+      )
+    }
+    return MutableLiveData(AsyncResult.failed(Throwable("Wrong exploration id")))
+  }
+
+  @Suppress("RedundantSuspendModifier") // DataProviders expects this function to be a suspend function.
+  private suspend fun retrieveWelcomeExplorationAsync(): AsyncResult<Exploration> {
+    try {
+      return AsyncResult.success(createExploration("welcome.json"))
+    } catch (e: Exception) {
+      return AsyncResult.failed(e)
+    }
+  }
+
+  @Suppress("RedundantSuspendModifier") // DataProviders expects this function to be a suspend function.
+  private suspend fun retrieveAbboutOppiaExplorationAsync(): AsyncResult<Exploration> {
+    try {
+      return AsyncResult.success(createExploration("about_oppia.json"))
+    } catch (e: Exception) {
+      return AsyncResult.failed(e)
+    }
+  }
+
+  // Returns an exploration given an assetName
+  private fun createExploration(assetName: String): Exploration {
+    try {
+      val explorationObject = loadJsonFromAsset(assetName) ?: return Exploration.getDefaultInstance()
+      return Exploration.newBuilder()
+        .setTitle(explorationObject.getString("title"))
+        .setLanguageCode(explorationObject.getString("language_code"))
+        .setInitStateName(explorationObject.getString("init_state_name"))
+        .setObjective(explorationObject.getString("objective"))
+        .putAllStates(createStatesFromJsonObject(explorationObject.getJSONObject("states")))
+        .build()
+    } catch (e: IOException) {
+      throw(Throwable("Failed to load and parse the json asset file. %s", e))
+    }
+  }
+
+  // Returns a JSON Object if it exists, else returns null
+  private fun getJsonObject(parentObject: JSONObject, key: String): JSONObject? {
+    return try {
+      parentObject.getJSONObject(key)
+    } catch (e: JSONException) {
```

Comment:
Prefer letting this throw since it'll then be propagated up to the UI.

Reply:
Will be resolved in #183.

Reply:
So the reason I am hiding this instead of propagating the exception is because in the dummy explorations there are cases where certain values can be null. For them this will throw an exception. I am using this function only in the places where that can happen.